axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/FixedAssetLineMoveServiceImpl.java

```diff
@@ -667,7 +667,9 @@ public void generateSaleMove(
       Account debitAccount = fixedAsset.getFixedAssetCategory().getDebtReceivableAccount();
       BigDecimal debitAmount = disposalAmount.add(creditAmountTwo);
 
-      if (creditAccountOne == null || creditAccountTwo == null || debitAccount == null) {
+      if (creditAccountOne == null
+          || (creditAccountTwo == null && creditAmountTwo.compareTo(BigDecimal.ZERO) > 0)
```

Comment:
We'll get an NPE here, what are you trying to do?
Maybe: || creditAccountTwo == null || creditAmountTwo.compareTo(BigDecimal.ZERO) > 0 ??